src/System.Private.CoreLib/shared/System/Globalization/CalendarData.Unix.cs

```diff
@@ -134,16 +134,103 @@ private static bool EnumDatePatterns(string localeName, CalendarId calendarId, C
             {
                 List<string> datePatternsList = callbackContext.Results;
 
-                datePatterns = new string[datePatternsList.Count];
                 for (int i = 0; i < datePatternsList.Count; i++)
                 {
-                    datePatterns[i] = NormalizeDatePattern(datePatternsList[i]);
+                    datePatternsList[i] = NormalizeDatePattern(datePatternsList[i]);
                 }
+
+                if (dataType == CalendarDataType.ShortDates)
+                    FixDefaultShortDatePattern(datePatternsList);
+
+                datePatterns = datePatternsList.ToArray();
             }
 
             return result;
         }
 
+        // FixDefaultShortDatePattern will convert the default short date pattern from using 'yy' to using 'yyyy'
+        // And will ensure the original pattern still exist in the list
+        private static void FixDefaultShortDatePattern(List<string> shortDatePatterns)
+        {
+            if (shortDatePatterns.Count == 0)
+                return;
+
+            string s = shortDatePatterns[0];
+
+            if (s.Length > 100)
+                return;
+
+            Span<char> modifiedPattern = stackalloc char[s.Length + 2];
+            int index = 0;
+
+            while (index < s.Length)
+            {
+                if (s[index] == '\'')
+                {
+                    do 
+                    {
+                        modifiedPattern[index] = s[index];
+                        index++;
+                    } while (index < s.Length && s[index] != '\'');
+
+                    if (index >= s.Length)
+                        return;                 
+                }
+                else if (s[index] == 'y')
+                {
+                    modifiedPattern[index] = 'y';
+                    break;
+                }
+
+                modifiedPattern[index] = s[index];
+                index++;
+            }
+
+            if (index >= s.Length - 1 || s[index + 1] != 'y')
```

Comment:
`index + 1 < s.Length` would make it easier to see there is no edge case here, if you increment index before it might also simplify the rest a little bit

Reply:
I think for code readability, what I have is better is you read it we are checking where the index now and if we crossed the boundaries. I  prefer to keep the way I wrote but let me know if you still feel strongly about it.

Reply:
for this one not strongly